### PR TITLE
plan: fix the order of schema columns

### PR DESCRIPTION
**Pull request by hanfei1991**
We should reset the schema in physical optimization.

@zz-jason @winoros @lamxTyler @zimulala  PTAL

### CODE REVIEW

File: plan/new_physical_plan_builder.go [hunk 1]
```diff
@@ -47,7 +47,6 @@ func (p *requiredProp) enforceProperty(task task, ctx context.Context, allocator
 	for _, col := range p.cols {
 		sort.ByItems = append(sort.ByItems, &ByItems{col, p.desc})
 	}
-	sort.SetSchema(task.plan().Schema())
```
Comment (winoros) [inline plan/new_physical_plan_builder.go]: It's okay to reserve the schema set process here?

**Comment by winoros** [comment]
mysql-test failed

**Comment by hanfei1991** [comment]
@winoros fixed

**Comment by hanfei1991** [comment]
@zz-jason PTAL

**Comment by zz-jason** [comment]
fix https://github.com/pingcap/tidb/issues/3914

**Comment by zz-jason** [comment]
@winoros @lamxTyler @shenli PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**